src/driver/amdxdna/aie2_pci.c

```diff
@@ -54,11 +78,56 @@ static inline void aie2_dump_chann_info_debug(struct amdxdna_dev_hdl *ndev)
 	XDNA_DBG(xdna, "x2i ringbuf 0x%x", ndev->mgmt_x2i.rb_start_addr);
 	XDNA_DBG(xdna, "x2i rsize   0x%x", ndev->mgmt_x2i.rb_size);
 	XDNA_DBG(xdna, "x2i chann index 0x%x", ndev->mgmt_chan_idx);
+	if (!ndev->mgmt_prot_major)
+		return;
+
+	XDNA_DBG(xdna, "mailbox protocol major 0x%x", ndev->mgmt_prot_major);
+	XDNA_DBG(xdna, "mailbox protocol minor 0x%x", ndev->mgmt_prot_minor);
+}
+
+static bool aie2_is_legacy_mgmt_info(struct amdxdna_dev_hdl *ndev, u32 off)
+{
+	u32 *mgmt_info_bar_addr = ndev->sram_base + off;
+	u32 magic;
+
+	magic = readl(mgmt_info_bar_addr + MAGIC_OFFSET);
+	if (strcmp((const char *)&magic, MGMT_MBOX_MAGIC))
```

Comment:
strcmp? Why not just define MGMT_MBOX_MAGIC as u32 and do "=="?

Reply:
Thanks. Fixed.